chia/consensus/multiprocess_validation.py

```diff
@@ -69,24 +103,24 @@ def pre_validate_block(
             # no longer need prev_generators
             assert prev_generators is not None
             assert block.transactions_info is not None
-            block_generator = BlockGenerator(block.transactions_generator, prev_generators)
-            assert block_generator.program == block.transactions_generator
-            npc_result = get_name_puzzle_conditions(
-                block_generator,
-                min(constants.MAX_BLOCK_COST_CLVM, block.transactions_info.cost),
-                mempool_mode=False,
-                height=block.height,
-                constants=constants,
-            )
-            if npc_result.error is not None:
+
+            if block.transactions_info.cost > constants.MAX_BLOCK_COST_CLVM:
                 validation_time = time.monotonic() - validation_start
                 return PreValidationResult(
-                    uint16(npc_result.error), None, npc_result.conds, False, uint32(validation_time * 1000)
+                    uint16(Err.BLOCK_COST_EXCEEDS_MAX.value), None, None, uint32(validation_time * 1000)
                 )
```

Comment:
This condition is here instead of the current min(constants.MAX_BLOCK_COST_CLVM, block.transactions_info.cost)-expression, when running the block.